cranelift-codegen/meta/src/isa/x86/legalize.rs

```diff
@@ -463,6 +465,25 @@ pub(crate) fn define(shared: &mut SharedDefinitions, x86_instructions: &Instruct
         );
     }
 
+    // SIMD vall_true
+    let zeroes = constant(vec![0x00; 16]);
+    let eq = Literal::enumerator_for(&imm.intcc, "eq");
+    for ty in ValueType::all_lane_types().filter(allowed_simd_type) {
+        let vall_true = vall_true.bind(vector(ty, sse_vector_size));
+        let lane_type_as_int = LaneType::int_from_bits(ty.lane_bits() as u16);
+        let raw_bitcast_to_int = raw_bitcast.bind(vector(lane_type_as_int, sse_vector_size));
+        narrow.legalize(
+            def!(y = vall_true(x)),
+            vec![
+                def!(a = vconst(zeroes)),
+                def!(b = raw_bitcast_to_int(x)),
+                def!(c = icmp(eq, b, a)),
+                def!(d = x86_ptest(c, c)),
```

Comment:
Yes, and I think we can do somethin like this as a part of #1015. I didn't use `PMOVMSKB` here for the reason mentioned earlier (MSB vs LSB) but also it would involve an extra instruction or two at the end to convert the extracted `11111...` to Wasm's expected `...00001`.